cmd/controller/app/options/options.go

```diff
@@ -258,9 +261,8 @@ func (s *ControllerOptions) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.Namespace, "namespace", defaultNamespace, ""+
 		"If set, this limits the scope of cert-manager to a single namespace and ClusterIssuers are disabled. "+
 		"If not specified, all namespaces will be watched")
-	fs.BoolVar(&s.LeaderElect, "leader-elect", true, ""+
```

Comment:
Thanks for the comments.   Regarding to the CA part in isito, we actualy want to support multi signers and each of them might have different root CA. So we can not use the way you suggested as it only supports one signle root CA :)

Reply:
Thanks for the detailed suggestion! will update soon the PR soon :)

Reply:
Ouch, why do we need to remove `--leader-elec`? It is very useful for local dev 😅 It should never be turned off in prod, many we can add a comment that says why it exists and why anyone would want to disable it (e.g. for local testing?)

Reply:
> Thanks for the comments. Regarding to the CA part in isito, we actualy want to support multi signers and each of them might have different root CA. So we can not use the way you suggested as it only supports one signle root CA :)

It is surprising to me that istiod only supports a single root CA, is that a hard limitation? For istiod to discover the CA from the CSR is concerning to me as it ripe for maliciously changing the CA. What happens if the CA changes, does istiod replace the existing CA, add it to the exiting CA bundle?

While exposing the CA certificate isn't necessarily _bad_, it does allow users to subtly set-up a PKI which is insecure as workloads will blindly trust whatever is returned from the issuer.

Supporting istio is definitely a priority for the project so we should make sure it works. I would like to hear from others what they think about adding this PR, or something like it.

@jetstack/team-cert-manager

Reply:
@SgtCoDFish @wallrj  @munnerz @irbekrm @jakexks @inteon @RinkiyaKeDad

Reply:
I think one thing that stands out to me here is that there are two very distinct bits to this PR.

There's the hiding of the leader election flag, and there's the new `full-cert-chain` flag for controlling what appears in chains we output. This thread will become unreadable pretty quickly if there are discussions about two completely different changes going on.

I think to avoid confusion, it might be worthwhile splitting this into two PRs so discussion can be focused on each part. This PR should IMO be for the `leader-elect` flag, and a new PR should be raised for the `full-cert-chain` flag.

Is that reasonable?

Reply:
FWIW, since the discussion on adding CA data has already begun here, I would rather the new PR be the leader elect one. That way we don't loose any discussion on the CA data topic which I think is more critical.

Reply:
sure, that sounds good to me :+1: